packages/bierzo-wallet/src/logic/transactions.unit.spec.ts

```diff
@@ -31,19 +31,19 @@ withChainsDescribe('Logic :: transaction subscriptions', () => {
     const store = aNewStore();
     const keys = await createPubkeys();
 
-    await drinkFaucetIfNeeded(keys);
     await subscribeTransaction(keys, store.dispatch);
 
-    // Trick for forcing account to receive balance events updates
+    // Trigger incoming SendTransactions
     await drinkFaucetIfNeeded(keys);
 
-    // Give some time to open request to be finished
-    await sleep(5000);
-    await sleep(5000);
+    // Wait for events to be processed
+    await sleep(1000);
 
-    // 2 (cash) + 2 (bash) + 2 eth
-    expect(txsSpy).toHaveBeenCalledTimes(6);
+    // Got one incoming transaction for BASH, CASH, ETH
+    expect(txsSpy).toHaveBeenCalledTimes(3);
+    const transactions = txsSpy.mock.calls.map(call => call[0]);
+    expect(new Set(transactions.map(tx => tx.amount.tokenTicker))).toEqual(new Set(['BASH', 'CASH', 'ETH']));
```

Comment:
Why can't we simply check against array and not against `Set`?

Reply:
Because we don't know the oder in which the transaction events are coming in. The arrays `["BASH", "CASH", "ETH"]` and `["ETH", "BASH", "CASH"]` are different. But when converted to a set, the elements have no order anymore, such that `Set(["BASH", "CASH", "ETH"])` and `Set(["ETH", "BASH", "CASH"])` are equal.